velox/functions/prestosql/json/JsonExtractor.cpp

```diff
@@ -122,6 +122,7 @@ void extractArray(
     auto rv = folly::tryTo<int32_t>(key);
     if (rv.hasValue()) {
       auto idx = rv.value();
+      idx = idx >= 0 ? idx : arrayLen + idx;
```

Comment:
nit:  arrayLen + idx wont overflow I suppose .

Reply:
It wont, arrayLen will always be non-negative and the addition is only done when idx is negative.
However, I did notice that arrayLen is size_t and idx is int32_t which can brittle when it comes to implicit conversions applied by the compiler. Therefore, I have explicitly changes both to int64_t.